### PR TITLE
feature: Log Server messages to VS Code output panel #1949

### PR DESCRIPTION
**Pull request by amanPras**
## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Check server logs are visible under the output terminal of the servers.
- [x] Check that the multiple instances of the client don't result in any mix and match-of logs.

## Checklist:
- [x] Each of my commits contains one meaningful change
- [x] I have performed rebase of my branch on top of the development
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -312,7 +312,7 @@ To generate a troubleshooting log for the LSP server, specify a value for the pa
 
 These values are ordered from returning the least information ("ERROR"; errors only) to the most information ("ALL"; all details).
 
-Troubleshooting logs are stored in the subfolder **\LSPCobol\logs** within your user home directory.
+Server logs could be found under "COBOL Language Support" output terminal.
```
Comment (amanPras) [inline README.md]: @zeibura Please check the update in README

**Review by zeibura — APPROVED** [review]

**Comment by zeibura** [comment]
I'll fix the readme later, can't edit forks

File: README.md (hunk 1 above)
Comment (VitGottwald) [reply README.md]: Should be called either just "output" or "output channel" according to VS Code terminology.

**Comment by VitGottwald** [comment]
> I'll fix the readme later, can't edit forks

That should be possible to fix. @ap891843 could you, please, check https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork and test it with @zeibura ?

**Comment by zeibura** [comment]
> > I'll fix the readme later, can't edit forks
> 
> That should be possible to fix. @ap891843 could you, please, check https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork and test it with @zeibura ?

No, please, dont normalise this practice. Instead, when there needs to be an update to the documentation, just tell me and I will do it on a separate PR. Its easier that way.

**Comment by zeibura** [comment]
I can edit the fork now - fixed the readme.

**Review by Nurkambay — CHANGES_REQUESTED** [review]
Please convert all `communications.logGeneralMessage` statements to `log.debug` statements (see `AnalysisService`)

**Comment by amanPras** [comment]
> Please convert all `communications.logGeneralMessage` statements to `log.debug` statements (see `AnalysisService`)

Updated

**Review by Nurkambay — APPROVED** [review]

**Pull request merged**